doc/Overview/Validation/ValidationOverview.md

```diff
@@ -1,16 +1,158 @@
+        {
+            // Handle validation errors
+        }
+    }
+}
+```
+
+The service checks whether the `Person` object is valid and handles calling the correct validator registered for the object type. It returns a `ValueTask` that completes when validation is complete. The `ValueTask` result is a `IEnumerable<ValidationResult>` which contains elements which indicate whether the object is valid or not. Uno Extensions enables fine-grained control over how validation is performed by allowing systematic use of a number of patterns from the .NET ecosystem. This is done by offering customization of both the validatable entity and within a validator service itself.
+
+## Validatable entities
+
+Instances of a class which are passed into the `ValidateAsync` method of the default validator have three primary implementation options to allow for validation:
+
+# [**INotifyDataErrorInfo**](#tab/notify-data-error-info)
+A common choice for entities that need to implement validation is the `INotifyDataErrorInfo` interface. Internally, the `GetErrors` method exposed by these objects is called by the validator. The .NET Community Toolkit provides an `ObservableValidator` class that implements this interface and provides the base functionality to execute validation rules on properties that have been decorated with validation attributes. `ObservableValidator` is a good choice for objects that need to be observable because it also implements `ObservableObject`. 
+
+### Code example
+
+An example using validation attributes is shown below:
 
-## Reference
+```csharp
+public class Person : ObservableValidator
+{
+    [Required]
+    public string FirstName { get; set; }
 
-- [For more information on FluentValidation](https://fluentvalidation.net/).
+    [Required]
+    public string LastName { get; set; }
+
+    [Range(0, 100)]
+    public int Age { get; set; }
+}
+```
+
+It is also possible to define custom validation attributes by deriving from the `ValidationAttribute` class. More information on custom validation can be found [here](https://nicksnettravels.builttoroam.com/custom-validation/).
+
+For more information, see the reference documentation about [INotifyDataErrorInfo](https://learn.microsoft.com/dotnet/api/system.componentmodel.inotifydataerrorinfo) and [ObservableValidator](https://docs.microsoft.com/windows/communitytoolkit/mvvm/observablevalidator).
+
+# [**IValidatableObject**](#tab/validatable-object)
+Another option that was developed more recently is the `IValidatableObject` interface. Like `INotifyDataErrorInfo`, it is implemented by a class to ensure properties can be validated based on the attributes they are decorated with. Validator support for this interface is provided internally with the `TryValidateObject` method used by the default validator to execute validation rules on the entity in question. This method returns an `IEnumerable<ValidationResult>` which contains elements which indicate whether the object is valid or not. 
```

Comment:
```suggestion
Another option is the `IValidatableObject` interface. Like `INotifyDataErrorInfo`, it is implemented by a class to ensure properties can be validated based on the attributes they are decorated with. Validator support for this interface is provided internally with the `TryValidateObject` method used by the default validator to execute validation rules on the entity in question. This method returns an `IEnumerable<ValidationResult>` which contains elements which indicate whether the object is valid or not. 
```

`IValidatableObject` has been there long enough to be non-recent.  
As far as I can remember I was annoyed by Silverlight's lack of support for this interface I have been using quite often...